### PR TITLE
Use lower case name for Logger variables

### PR DESCRIPTION
**Pull request by kokosing**
Use lower case name for Logger variables

### CODE REVIEW

**Comment by martint** [comment]
On the contrary, they should be capitalized since they are “final static”

**Comment by findepi** [comment]
@martint  Some would argue that ALL_CAPS is constants (and constants should be `static final`) and some `static` fields that are not constants should be `camelCase`. And a Logger is no constant.

Current practice in our code base seems to either follow the above, or treat `log` as an exception of the rule:
```
$ git grep 'private static final Logger' | cut -d: -f2 | awk '{print $5}' | sort | uniq -c | sort -n
      4 LOGGER
     11 LOG
    136 log
```

File: presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java [hunk 1]
```diff
@@ -92,7 +92,7 @@
 @ThreadSafe
 public class QueryStateMachine
 {
-    public static final Logger QUERY_STATE_LOG = Logger.get(QueryStateMachine.class);
+    public static final Logger log = Logger.get(QueryStateMachine.class);
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java]: Revert this change. This is not standard class-level logger. It's shared between 2 classes.

Note: I am not arguing this is a good idea to share a logger between classes. However, as long as it's shared, it deserves special name.

Also, you can make it package private at least.

**Comment by martint** [comment]
> @martint Some would argue that ALL_CAPS is constants (and constants should be static final) and some static fields that are not constants should be camelCase. And a Logger is no constant.

I guess it depends on your definition of constant. It's a global non-varying reference to an object that has no observable mutable state, so that seems pretty constant to me. It's no different than, say, `BigintType.BIGINT`, `VarcharType.VARCHAR` and other common uses of `INSTANCE` to represent a singleton object.

I've been opportunistically and intentionally using the capitalized form on any new code I write, so making them lower case will undo that.

**Comment by martint** [comment]
From a practical standpoint, requiring that all `static final` fields be named using all caps makes it easier to validate and enforce via a checkstyle rule if we wanted to go that route. Having that rule be based on whether the object referenced by the field is immutable or not makes that automation impossible. 

In the absence of an automated check, it'd be very easy for a developer to get it wrong, too. Sometimes, it's not immediately obvious if an object is immutable. Or a simple refactoring that changes the initialization of the field to create an immutable version of an object that was previously mutable might go unnoticed.

**Comment by kokosing** [comment]
I am closing this. Changing the code other way round is not that easy to be done automatically.
